src/app/(sidebar)/transaction/sign/page.tsx

```diff
@@ -1,5 +1,198 @@
+      setIsTxImported(false);
+      setTxErrMsg("Unable to import a transaction envelope");
+    }
+  };
+
+  const rendeImportView = () => {
+    return (
+      <>
+        <Card>
+          <div className="SignTx__xdr">
+            <XdrPicker
+              id="sign-transaction-xdr"
+              label={
+                <Text size="xs" as="span" weight="medium">
+                  Import a transaction envelope in XDR format
+                  <span className="SignTx__icon">
+                    <Icon.AlertCircle />
+                  </span>
+                </Text>
+              }
+              value={txEnv || ""}
+              error={txErrMsg}
+              note={
+                <Text size="xs" as="span" addlClassName="success-message">
+                  {txSuccessMsg}
+                </Text>
+              }
+              onChange={(e) => onChange(e.target.value)}
+            />
+
+            <div className="SignTx__CTA">
+              <Button
+                disabled={!txEnv || !isTxValid}
+                size="md"
+                variant={"secondary"}
+                onClick={onImport}
+              >
+                Import transaction
+              </Button>
+            </div>
+          </div>
+        </Card>
+
+        <Alert
+          placement="inline"
+          variant="primary"
+          actionLink="https://developers.stellar.org/network/horizon/resources"
+          actionLabel="Read more about signatures on the developer's site"
+        >
+          <Text size="sm" as="p">
+            The transaction signer lets you add signatures to a Stellar
+            transaction. Signatures are used in the network to prove that the
+            account is authorized to perform the operations in the transaction.
+          </Text>
+          <Text size="sm" as="p">
+            For simple transactions, you only need one signature from the
+            correct account. Some advanced transactions may require more than
+            one signature if there are multiple source accounts or signing keys.
+          </Text>
+        </Alert>
+      </>
+    );
+  };
+
+  const renderOverviewView = () => {
+    if (!tx) {
+      return;
+    }
+
+    const REQUIRED_FIELDS = [
+      {
+        label: "Signing for",
+        value: network.passphrase,
+      },
+      {
+        label: "Transaction Envelope XDR",
+        value: txEnv,
+      },
+      {
+        label: "Transaction Hash",
+        value: tx.hash().toString("hex"),
+      },
+    ];
+
+    let mergedFields;
+
+    if (tx instanceof FeeBumpTransaction) {
+      mergedFields = [...REQUIRED_FIELDS, ...FEE_BUMP_TX_FIELDS(tx)];
+    } else {
+      mergedFields = [...REQUIRED_FIELDS, ...TX_FIELDS(tx)];
+    }
+
+    return (
+      <>
+        <Card>
+          <div className="SignTx__FieldViewer">
+            {mergedFields.map((field) => {
+              const className =
+                field.value.toString().length >= MIN_LENGTH_FOR_FULL_WIDTH_FIELD
```

Comment:
Are we always guaranteed to have `field.value`? Just making sure we don't need to chain these like optional with `?`.

Reply:
yes!